src/battle_main.c

```diff
@@ -322,85 +322,83 @@ const u8 gTypeNames[NUMBER_OF_MON_TYPES][TYPE_NAME_LENGTH + 1] =
     [TYPE_FAIRY] = _("Fairy"),
 };
 
-#define DEFAULT_MONEY 5
-#define DEFAULT_BALL ITEM_POKE_BALL
-
-#define TRAINER_CLASS(trainerClass, trainerName, trainerMoney, trainerBall) \
-    [TRAINER_CLASS_##trainerClass] =                                        \
-    {                                                                       \
-        .name = _(trainerName),                                             \
-        .money = trainerMoney,                                              \
-        .ball = trainerBall,                                                \
+// extra args are money and ball
+#define TRAINER_CLASS(trainerClass, trainerName, ...)   \
+    [TRAINER_CLASS_##trainerClass] =                    \
+    {                                                   \
+        .name = _(trainerName),                         \
+        .money = DEFAULT(5, __VA_ARGS__),               \
+        .ball = DEFAULT_2(ITEM_POKE_BALL, __VA_ARGS__), \
     }
```

Comment:
It's such a shame that agbcc always throws an error if we specify two of the same member (and thus we're basically forced into `-Woverride-init`), I think the whole codebase would get a lot of mileage out of being able to specify default members of structs that only apply if those members aren't explicitly specified.

Not really a review comment, just something that might spark some inspiration in a future reader who can then make a follow-up PR with such a feature :)

Reply:
For now, I think `DEFAULT` is the best solution I can come up with...